From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matt Campbell <mattcampbell@pobox.com>
Date: Mon, 27 Mar 2023 16:02:16 -0500
Subject: fix: Allow old-style media constraints to specify channel count

Necessary for stereo audio capture in RIM.

diff --git a/third_party/blink/renderer/modules/mediastream/media_constraints_impl.cc b/third_party/blink/renderer/modules/mediastream/media_constraints_impl.cc
index 707695559e2c6c9dca9399f4045279c60d213b36..8283196f2f7590f1ecb2579251cfec951f1d9cb8 100644
--- a/third_party/blink/renderer/modules/mediastream/media_constraints_impl.cc
+++ b/third_party/blink/renderer/modules/mediastream/media_constraints_impl.cc
@@ -87,6 +87,7 @@ const char kMaxHeight[] = "maxHeight";
 const char kMinHeight[] = "minHeight";
 const char kMaxFrameRate[] = "maxFrameRate";
 const char kMinFrameRate[] = "minFrameRate";
+const char kChannelCount[] = "channelCount";
 const char kMediaStreamSource[] = "chromeMediaSource";
 const char kMediaStreamSourceId[] =
     "chromeMediaSourceId";                           // mapped to deviceId
@@ -299,6 +300,8 @@ static void ParseOldStyleNames(
       result.frame_rate.SetMin(atof(constraint.value_.Utf8().c_str()));
     } else if (constraint.name_.Equals(kMaxFrameRate)) {
       result.frame_rate.SetMax(atof(constraint.value_.Utf8().c_str()));
+    } else if (constraint.name_.Equals(kChannelCount)) {
+      result.channel_count.SetExact(atoi(constraint.value_.Utf8().c_str()));
     } else if (constraint.name_.Equals(kEchoCancellation)) {
       result.echo_cancellation.SetExact(ToBoolean(constraint.value_));
     } else if (constraint.name_.Equals(kMediaStreamSource)) {
